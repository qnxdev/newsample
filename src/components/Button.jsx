import { Component } from "react";

class Button extends Component {
  render() {
    function fn2() {
      console.log("a,b");
    }
    const { a, b, fn, children } = this.props;
    return (
      <div onClick={fn} className="Button">
        {a}
        {b}
      </div>
    );
  }
}

export default Button;
