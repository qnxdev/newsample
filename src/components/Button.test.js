import Button from "./Button";

import { shallow } from "enzyme";
import toJson from 'enzyme-to-json';

describe("Button", () => {
  it("renders without any error", function () {
    const component = shallow(<Button />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

});