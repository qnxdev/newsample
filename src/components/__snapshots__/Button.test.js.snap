// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders without any error 1`] = `
<button
  className="Button"
/>
`;
